cmd/xl-storage-disk-id-check.go

```diff
@@ -665,10 +666,21 @@ func scannerTrace(s scannerMetric, startTime time.Time, duration time.Duration,
 }
 
 // Update storage metrics
-func (p *xlStorageDiskIDCheck) updateStorageMetrics(s storageMetric, paths ...string) func(err *error) {
+func (p *xlStorageDiskIDCheck) updateStorageMetrics(s storageMetric, ctx context.Context, paths ...string) func(err *error) {
 	startTime := time.Now()
 	trace := globalTrace.NumSubscribers(madmin.TraceStorage) > 0
+	incS3Count := false
+
+	if tc, ok := ctx.Value(mcontext.ContextTraceKey).(*mcontext.TraceCtxt); ok && strings.HasPrefix(tc.FuncName, "s3.") {
```

Comment:
>  strings.HasPrefix(tc.FuncName, "s3.")

This doesn't look robust enough for now; maybe we need to set a new context value that can be spread between nodes using another header

Reply:
Not needed. The health tracker already tracks ongoing operations and there is no magic there.